esp_encrypted_img/src/esp_encrypted_img.c

```diff
@@ -144,6 +147,91 @@ static int decipher_gcm_key(const char *enc_gcm, esp_encrypted_img_t *handle)
 
     return (ret);
 }
+
+static esp_err_t esp_encrypted_img_export_rsa_pub_key(const char *rsa_pem, size_t rsa_len, uint8_t **pub_key, size_t *pub_key_len)
+{
+    int ret = 0;
+    if (rsa_pem == NULL) {
+        ESP_LOGE(TAG, "RSA private key is not set");
+        return ESP_ERR_INVALID_ARG;
+    }
+    mbedtls_pk_context pk;
+    mbedtls_pk_init(&pk);
+
+    mbedtls_entropy_context entropy;
+    mbedtls_ctr_drbg_context ctr_drbg;
+
+    mbedtls_ctr_drbg_init( &ctr_drbg );
+    mbedtls_entropy_init( &entropy );
+
+#if (MBEDTLS_VERSION_NUMBER < 0x03000000)
+    if ( (ret = mbedtls_pk_parse_key(&pk, (const unsigned char *) rsa_pem, rsa_len, NULL, 0)) != 0) {
+#else
+    if ( (ret = mbedtls_pk_parse_key(&pk, (const unsigned char *) rsa_pem, rsa_len, NULL, 0, mbedtls_ctr_drbg_random, &ctr_drbg)) != 0) {
+#endif
+        ESP_LOGE(TAG, "failed\n  ! mbedtls_pk_parse_key returned -0x%04x\n", (unsigned int) - ret );
+        goto exit;
+    }
+
+    if (mbedtls_pk_get_type(&pk) != MBEDTLS_PK_RSA) {
+        ESP_LOGE(TAG, "Public key is not RSA");
+        goto exit;
+    }
+
+    const mbedtls_rsa_context *rsa_ctx = mbedtls_pk_rsa(pk);
+    if (rsa_ctx == NULL) {
+        ESP_LOGE(TAG, "Failed to get RSA context from public key");
+        goto exit;
+    }
+
+    size_t max_pub_key_size = MBEDTLS_MPI_MAX_SIZE + RSA_MPI_ASN1_HEADER_SIZE;
+    *pub_key = calloc(1, max_pub_key_size + 1);
+    if (*pub_key == NULL) {
+        ESP_LOGE(TAG, "Failed to allocate memory for public key");
+        goto exit;
+    }
+
+    unsigned char *c = *pub_key + max_pub_key_size;
+
+    ret = mbedtls_pk_write_pubkey(&c, *pub_key, &pk);
+    if (ret < 0) {
+        ESP_LOGE(TAG, "Failed to write public key: -0x%04x", (unsigned int) - ret);
+        goto exit;
+    }
+    if (c - *pub_key < 0) {
```

Comment:
If the public key was successfully written above, why would the public key buffer too small here?